driver/src/gnosis_safe_gas_station.rs

```diff
@@ -0,0 +1,86 @@
+use ethcontract::U256;
+use isahc::prelude::*;
+use serde::Deserialize;
+use std::time::Duration;
+use uint::FromDecStrErr;
+
+/// Result of https://safe-relay.gnosis.io/ api call `/v1/gas-station`.
+///
+/// Prices are in wei.
+#[derive(Deserialize, Debug, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct GasPrice {
+    pub last_update: String,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub lowest: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub safe_low: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub standard: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub fast: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub fastest: U256,
+}
+
+/// Retrieves gas prices from the Gnosis Safe Relay api.
+///
+/// Uses a timeout of 10 seconds.
+pub fn get_gas_price() -> Result<GasPrice, isahc::Error> {
+    const URL: &str = "https://safe-relay.gnosis.io/api/v1/gas-station/";
+    const TIMEOUT: Duration = Duration::from_secs(10);
+    // It would be more efficient to reuse the client between calls. However, we
+    // only call this function once per batch when submitting a solution so this
+    // is not important at the moment.
+    let client = HttpClient::builder().timeout(TIMEOUT).build()?;
```

Comment:
Can we make the gas station a struct and pass this from above to the contract? Even though performance is not a main concern I think this is only a little bit of extra work to make the factoring here clean (the gas estimation is a dependency of the contract and should thus be injected and initializable with e.g. a mocked estimator).

Reply:
I made this mockable by introducing a struct and trait and passing it down.